### PR TITLE
chore: remove support for old .forceignore file format

### PR DESCRIPTION
**Pull request by cristiand391**
### What issues does this PR fix or reference?

@W-9155917@

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

File: src/resolve/forceIgnore.ts [hunk 1]
```diff
@@ -37,17 +28,9 @@ export class ForceIgnore {
     try {
       this.contents = readFileSync(forceIgnorePath, 'utf-8');
       // add the default ignore paths, and then parse the .forceignore file
-      // DO NOT CALL parseContents FOR THE NEW PARSER
-      // the new library handles it's own unix/windows file path separators, let it handle it
       this.parser = ignore().add(`${this.contents}\n${this.DEFAULT_IGNORE.join('\n')}`);
-      // TODO: START REMOVE AFTER GITIGNORE-PARSER DEPRECATED
-      // add the default and send to the old gitignore-parser
-      this.gitignoreParser = gitignoreParser.compile(this.parseContents(this.contents));
+      this.contents = this.contents.split('\n').join('\n');
```
Comment (cristiand391) [inline src/resolve/forceIgnore.ts:32]: This is just a workaround, if you remove this line then [this test will fail](https://github.com/forcedotcom/source-deploy-retrieve/blob/a68cb515202761d290548cae7a397fe29e541d2c/test/client/retrieve.test.ts#L148).

When `forceIgnorePath == ''`, line 45 would throw an error when passing `this.contents` as `undefined` to `this.parseContents` and would leave `this.forceIgnoreDirectory` as `undefined`.

File: test/resolve/forceIgnore.test.ts [hunk 2]
```diff
@@ -204,41 +98,4 @@ describe('ForceIgnore', () => {
...
-      expect(forceIgnore.denies(dupPath)).to.be.false;
-    });
-
-    it('Should ignore files named package2-descriptor.json', () => {
-      const descriptorPath = join(root, 'package2-descriptor.json');
-      expect(forceIgnore.accepts(descriptorPath)).to.be.true;
-      expect(forceIgnore.denies(descriptorPath)).to.be.false;
-    });
-
-    it('Should ignore files named package2-manifest.json', () => {
```
Comment (WillieRuemmele) [inline test/resolve/forceIgnore.test.ts:238]: These should stay because they're testing the default entries

Comment (cristiand391) [reply test/resolve/forceIgnore.test.ts:238]: @WillieRuemmele these were only testing the old parser, the same tests are still present: https://github.com/forcedotcom/source-deploy-retrieve/blob/dee6f33901df78b886f2c7e06480ce4618d33826/test/resolve/forceIgnore.test.ts#L66-L98

**Review by WillieRuemmele — APPROVED** [review]

**Review by mshanemc — APPROVED** [review]

**Comment by WillieRuemmele** [comment]
verified TB NUTs pass too
verified with push/pull/status
plugin-source NUTs "passing"

**Pull request merged**